diff --git a/node_modules/napajs/lib/zone/zone-impl.js b/node_modules/napajs/lib/zone/zone-impl.js
index 209b01b..880daaf 100644
--- a/node_modules/napajs/lib/zone/zone-impl.js
+++ b/node_modules/napajs/lib/zone/zone-impl.js
@@ -58,12 +58,14 @@ var ZoneImpl = /** @class */ (function () {
         var source = this.createBroadcastSource(arg1, arg2);
         return new Promise(function (resolve, reject) {
             _this._nativeZone.broadcast(source, function (resultCode) {
-                if (resultCode === 0) {
-                    resolve();
-                }
-                else {
-                    reject("broadcast failed with result code: " + resultCode);
-                }
+                setImmediate(function () {
+                    if (resultCode === 0) {
+                        resolve();
+                    }
+                    else {
+                        reject("broadcast failed with result code: " + resultCode);
+                    }
+                });
             });
         });
     };
@@ -72,12 +74,14 @@ var ZoneImpl = /** @class */ (function () {
         var spec = this.createExecuteRequest(arg1, arg2, arg3, arg4);
         return new Promise(function (resolve, reject) {
             _this._nativeZone.execute(spec, function (result) {
-                if (result.code === 0) {
-                    resolve(new Result(result.returnValue, transport.createTransportContext(true, result.contextHandle)));
-                }
-                else {
-                    reject(result.errorMessage);
-                }
+                setImmediate(function () {
+                    if (result.code === 0) {
+                        resolve(new Result(result.returnValue, transport.createTransportContext(true, result.contextHandle)));
+                    }
+                    else {
+                        reject(result.errorMessage);
+                    }
+                });
             });
         });
     };
